domain/cloudimagemetadata/state/state.go

```diff
@@ -0,0 +1,381 @@
+	declareInClause := func(field string, values []string) {
+		if len(values) != 0 {
+			for i, value := range values {
+				values[i] = fmt.Sprintf(`'%s'`, value)
+			}
+			clauses = append(clauses, fmt.Sprintf(`%s IN (%s)`, field,
+				strings.Join(values, ", ")))
+		}
+	}
+
+	declareEqualsClause("cloud_image_metadata.region", criteria.Region)
+	declareInClause("cloud_image_metadata.version", criteria.Versions)
+	declareInClause("arch.architecture_name", criteria.Arches)
+	declareEqualsClause("cloud_image_metadata.stream", criteria.Stream)
+	declareEqualsClause("cloud_image_metadata.virt_type", criteria.VirtType)
+	declareEqualsClause("cloud_image_metadata.root_storage_type", criteria.RootStorageType)
+
+	findMetadataQuery := fmt.Sprintf(`
+	WITH arch(id, architecture_name) AS (
+		SELECT id, name FROM architecture
+	)
+	SELECT &outputMetadata.* FROM cloud_image_metadata
+	JOIN arch ON cloud_image_metadata.architecture_id = arch.id
+	WHERE %s;
+	`, strings.Join(clauses, ` AND `))
+
+	findMetadataStmt, err := sqlair.Prepare(findMetadataQuery, outputMetadata{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var metadatas []outputMetadata
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, findMetadataStmt).GetAll(&metadatas)
+		if dberrors.IsErrNotFound(err) {
+			return cloudmetadataerrors.NotFound
+		}
+		return err
+	})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	result := make([]cloudimagemetadata.Metadata, len(metadatas))
+	for i, m := range metadatas {
+		result[i] = cloudimagemetadata.Metadata{
+			MetadataAttributes: cloudimagemetadata.MetadataAttributes{
+				Stream:          m.Stream,
+				Region:          m.Region,
+				Version:         m.Version,
+				Arch:            m.ArchitectureName,
+				VirtType:        m.VirtType,
+				RootStorageType: m.RootStorageType,
+				RootStorageSize: m.RootStorageSize,
+				Source:          m.Source,
+			},
+			Priority:     m.Priority,
+			ImageId:      m.ImageID,
+			CreationTime: m.CreatedAt,
+		}
+	}
+	return result, nil
+}
+
+// AllCloudImageMetadata retrieves all cloud image metadata from the database without applying any filter criteria.
+// Returns a slice of [cloudimagemetadata.Metadata] and any error encountered during the retrieval process.
+func (s *State) AllCloudImageMetadata(ctx context.Context) ([]cloudimagemetadata.Metadata, error) {
+	result, err := s.FindMetadata(ctx, cloudimagemetadata.MetadataFilter{})
+	if errors.Is(err, cloudmetadataerrors.NotFound) {
+		return nil, nil
+	}
+	return result, errors.Trace(err)
+}
+
+// tryCleanUpExpiredMetadata removes metadata records from the cloud_image_metadata table that are older than [ExpirationDelay].
+//
+// Custom images doesn't expire.
+func (s *State) tryCleanUpExpiredMetadata(ctx context.Context) error {
+	db, err := s.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	deleteQuery := fmt.Sprintf(`
+DELETE FROM cloud_image_metadata 
```

Comment:
```suggestion
DELETE 
FROM cloud_image_metadata 
```

Reply:
I find it less readable with a new line (even i understand why you find it more consistent with SELECT)

My opinion is:
- The emphasis on Select is on both fields and table. It is good to have two line
```sql
SELECT <fields..>
FROM <tables...>
```
- The emphasis on DELETE is only on table. There is no need to have two lines
```sql
DELETE FROM <tables...>
```

Reply:
yeah you are right, I went a bit too far with those newlines :)